optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/common/accessor/gizmo/GizmoMemberAccessorFactory.java

```diff
@@ -36,8 +39,11 @@
      * @return The generated class name for member
      */
     public static String getGeneratedClassName(Member member) {
-        return member.getDeclaringClass().getPackage().getName() + ".$optaplanner$__"
-                + member.getDeclaringClass().getSimpleName() + "$__" + member.getName();
+        String memberName = Optional.ofNullable(ReflectionHelper.getGetterPropertyName(member))
+                .orElse(member.getName());
```

Comment:
nitpick: we typical avoid Optional in everything except for streams (which is also why it was designed for in the first place). This code creates an Optional instance every time to throw it away. 
`org.apache.commons.lang3.ObjectUtils.defaultIfNull` accomplishes the same goal, or just a `a == null ? ... : a`. Neither of those create a new object instance